src/Sarif.WorkItems/SarifWorkItemsExtensions.cs

```diff
@@ -103,20 +104,47 @@ private static string ConstructFullRuleIdentifier(ReportingDescriptor reportingD
             return fullRuleIdentifier;
         }
 
-        public static string CreateWorkItemDescription(this SarifLog log)
+        public static string CreateWorkItemDescription(this SarifLog log, Uri locationUri)
         {
-            Dictionary<string, int> resultCountsByTool = ComputeToolResultCounts(log);
-            StringBuilder templateText = new StringBuilder(@"This bug contains results for the below tool(s) and issue(s).  It was filed automatically.");
-            templateText.AppendLine();
-            templateText.AppendLine();
+            Dictionary<Run, int> resultCountsByRun = ComputeRunResultCounts(log);
+            StringBuilder templateText = new StringBuilder();
+            int runningResults = 0;
+            string toolNames = string.Empty;
+            string multipleToolsFooter = string.Empty;
+
+            Uri runRepositoryUri = resultCountsByRun.FirstOrDefault().Key?.VersionControlProvenance?.FirstOrDefault().RepositoryUri;
```

Comment:
>. [](start = 69, length = 1)

`?.` here as well, since the whole point of `FirstOrDefault` is that it returns `null` if the collection is empty.

Reply:
I'm sorry, I pointed my comment at the wrong place. It's the output of `FirstOrDefault()` that might be `null`, so you have to say `...FirstOrDefault()?.Key?.VersionControlProvenance?...`.

I see from looking at  `ComputeRunResultsCount` that `resultCountsByRun` _cannot_ be `null`, so you don't have to write `resultCountsByRun?.`.

Finally, if a dictionary entry does exist, then `Key` cannot be `null`.

So in summary: `resultCountsByRun.FirstOrDefault()?.Key.VersionControlProvenance?.FirstOrDefault()?.RepositoryUri`

Note the `?.` after _both_ calls to `FirstOrDefault()` in this expression.

---
In reply to: [398195637](https://github.com/microsoft/sarif-sdk/pull/1823#discussion_r398195637) [](ancestors = 398195637)

Reply:
I was wrong. This is correct as is. `KeyValuePair` is a value type, so can't be `null` -- but the `Key` _can_ be `null`, so your code is exactly right.

---
In reply to: [398745325](https://github.com/microsoft/sarif-sdk/pull/1823#discussion_r398745325) [](ancestors = 398745325,398195637)